core/shared/src/main/scala/scalaz/zio/ZSchedule.scala

```diff
@@ -637,8 +664,8 @@ private[zio] trait Schedule_Functions extends Serializable {
    * between steps. You can chain this onto the end of schedules to find out
    * what their delay is, e.g. `Schedule.spaced(1.second) >>> Schedule.delay`.
    */
-  final val delay: Schedule[Any, Duration] =
-    forever.reconsider[Any, Duration]((_, d) => d.copy(finish = () => d.delay))
+  final val delay: Schedule[Any, Delay] =
```

Comment:
We should probably add a `final val duration: Schedule[Any, Duration]` that uses `mapM` on `delay` to convert the `Delay` into a `Duration` since those are vastly easier to work with.

Reply:
@jdegoes I've added `duration` implementation as:

`final val duration: Schedule[Any, Duration] = delay.mapM(_.run)`

However, when testing this with `>>>`, I was always getting a zeroed `Delay`. Even testing with master branch I'm getting the same result.

When debugging, I found that `delay.reconsider` is happening before `>>>.combineWith`, always issuing a zeroed delay.

Reply:
This is my mistake. In order to make it work with `>>>` and the current type signature, we'd have to change `update` to take a "preceding" delay (basically threading it through the composition). 

Now, this change may independently be a good change. But I think it's a mistake to conflate it with this refactoring, we can always do it separately.

So what I would suggest is change `duration` to have this type:

```scala
final val duration: Schedule[Delay, Duration] = 
```

Then it can operate on the input to the `Schedule`, which will be a `Delay`.

We already have `fromFunction`, if we add `fromFunctionM`, then we can implement `duration` with that, and add a new useful combinator in the process.

Does all this make sense?

Reply:
I've come with this:

``` 
final val duration: Schedule[Delay, Duration] =
    forever.reconsider[Delay, Delay]((s, d) => d.rightMap(_ => s)).mapM(_.run)
```

Does it make sense?

Reply:
This looks good to me! 😄